gql/client.py

```diff
@@ -31,22 +32,33 @@ def __init__(
         assert not (
             type_def and introspection
         ), "Cannot provide introspection type definition at the same time."
-        if transport and fetch_schema_from_transport:
+
+        if type_def:
             assert (
                 not schema
-            ), "Cannot fetch the schema from transport if is already provided."
+            ), "Cannot provide type definition and schema at the same time."
+            warnings.warn(
+                "type_def is deprecated; use schema instead",
+                category=DeprecationWarning,
+            )
+            schema = type_def
+
         if introspection:
             assert (
                 not schema
             ), "Cannot provide introspection and schema at the same time."
             schema = build_client_schema(introspection)
-        elif type_def:
+
+        if isinstance(schema, str):
+            type_def_ast = parse(schema)
+            schema = cast(GraphQLSchema, build_ast_schema(type_def_ast))
```

Comment:
Why the cast? As far as I know, the [**build_ast_schema**](https://github.com/graphql-python/graphql-core/blob/3267c4c5c5ea06968beb59f16425de4a4b835ee1/src/graphql/utilities/build_ast_schema.py#L23) returns a `GraphQLSchema` type.

Reply:
I had a mypy error earlier. But now it seems it is not necessary anymore. Well spotted.